README.md

```diff
@@ -0,0 +1,95 @@
+# Layer0 Auth0 Proxy
+
+Sometimes, you want to put a sensitive web application behind a login wall.
+Sometimes, you don't want to write the authentication logic yourself.
+In this repository, we provide a proxy application that authenticates through Auth0 and can be easily inserted into any Terraform deployment.
+
+
+# Usage
+
+## An Example
+
+Let's discuss what a possible deployment might look like.
+
+1. A Layer0 environment in which all of the following resources will live.
+2. A sensitive application deployed to AWS.
+3. A private load balancer that sits in front of the sensitive application.
+4. The auth0-proxy application, also deployed to AWS.
+5. A public load balancer that sits in front of the auth0-proxy application.
+
+If we hand-wave away the specifics of the sensitive application (the "myapp" service in the coming example), a sample Terraform deployment of this whole system might look like this:
+
+```
+# main.tf
+
+provider "layer0" {
+  endpoint = "${var.endpoint}"
+  token = "${var.token}"
+  skip_ssl_verify = true
+}
+
+resource "layer0_environment" "demo" {
+  name = "demo"
+}
+
+resource "layer0_load_balancer" "myapp" {
+  name = "myapp"
+  environment = "${layer0_environment.demo.id}"
+  private = true
+
+  port {
+    host_port      = 80
+    container_port = 80
+    protocol       = "http"
+  }
+}
+
+resource "layer0_service" "myapp" {
+  name          = "myapp"
+  environment   = "${layer0_environment.demo.id}"
+  load_balancer = "${layer0_load_balancer.myapp.id}"
+  # and any other values that myapp needs
+}
+
+# Here's what we do in order to add the auth0-proxy:
+module "auth0" {
+  source = "github.com/quintilesims/auth0-proxy//terraform"
```

Comment:
This should only include required input variables; any optional ones (e.g. `auth0_domain`, `docker_image_tag`) should be taken out of this example I think.

Reply:
@zpatrick I agree with `docker_image_tag`, but if this tool is supposed to be open-source (and since the repo is public), I don't feel comfortable with posting our auth0 domain as a default value.

Reply:
That's fair, bad example, but I think the intent of the comment is still valid.

Reply:
Yeah, I gotcha. :+1:

Reply:
✅